src/decay.hpp

```diff
@@ -0,0 +1,137 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+#ifndef DECAY_H
+#define DECAY_H
+
+#include <initializer_list>
+#include <unordered_map>
+#include <vector>
+#include <string>
+
+#include <boost/core/demangle.hpp>
+
+namespace flexiblesusy {
+
+class Decay {
+public:
+   Decay(int, std::initializer_list<int>, double, std::string const&);
+   ~Decay() = default;
+   Decay(const Decay&) = default;
+   Decay(Decay&&) = default;
+   Decay& operator=(const Decay&) = default;
+   Decay& operator=(Decay&&) = default;
+
+   int get_initial_particle_id() const { return pid_in; }
+   const std::vector<int>& get_final_state_particle_ids() const {
+      return pids_out;
+   }
+   std::size_t get_final_state_size() const { return pids_out.size(); }
+
+   double get_width() const { return width; }
+   std::string get_proc_string() const { return proc_string; }
+   void set_width(double w) { width = w; }
+
+private:
+   int pid_in{0};
+   std::vector<int> pids_out{};
+   double width{0.};
+   std::string proc_string;
+};
+
+std::size_t hash_decay(const Decay& decay);
+
+class Decays_list {
+private:
+   using List_type = std::unordered_map<std::size_t, Decay>;
+public:
+   using iterator = List_type::iterator;
+   using const_iterator = List_type::const_iterator;
+
+   explicit Decays_list(int);
+   ~Decays_list() = default;
+   Decays_list(const Decays_list&) = default;
+   Decays_list(Decays_list&&) = default;
+   Decays_list& operator=(const Decays_list&) = default;
+   Decays_list& operator=(Decays_list&&) = default;
+
+   iterator begin() noexcept { return decays.begin(); }
+   const_iterator begin() const noexcept { return decays.begin(); }
+   const_iterator cbegin() const noexcept { return decays.cbegin(); }
+   iterator end() noexcept { return decays.end(); }
+   const_iterator end() const noexcept { return decays.end(); }
+   const_iterator cend() const noexcept { return decays.end(); }
+
+   std::size_t size() const noexcept { return decays.size(); }
+
+   void clear();
+   // @todo
```

Comment:
This sounds really strange! Which model did you use where this behaviour can be seen?
Should one see the effect in the output SLHA line
~~~
DECAY        35     1.94400445E+00   # hh(2) decays
~~~
?

Reply:
I'm no sure what are you referring to :)

Reply:
Sorry. :) The `@todo` item says, that you're seeing a difference in the Higgs (which one?) total width. I'd like to reproduce this strange behaviour but I don't know which model and which Higgs boson you looked at (and where to find the output).

Reply:
Sorry, this github feature could be improved. Would be better if on click it took me to a right line. About the comment: - I know it sounds like a joke but I'm not crazy. I spent few hours diagnosing it. As far as I can say, it's seen only in the output. The actual numeric value is not affected.

Reply:
I've tried to reproduce the issue, but failed. Running the MSSM w/ and w/o the patch
~~~.patch
index 6fb6ea130..db56f609a 100644
--- a/src/decay.cpp
+++ b/src/decay.cpp
@@ -73,7 +73,7 @@ void Decays_list::clear()
    total_width = 0.;
 }
 
-void Decays_list::set_decay(double width, std::initializer_list<int> pids_out, std::string proc_string)
+void Decays_list::set_decay(double width, std::initializer_list<int> pids_out, const std::string& proc_string)
 {
    const Decay decay(initial_pdg, pids_out, width, proc_string);
    const auto decay_hash = hash_decay(decay);
diff --git a/src/decay.hpp b/src/decay.hpp
index c4f44823f..4fffa93a6 100644
--- a/src/decay.hpp
+++ b/src/decay.hpp
@@ -86,7 +86,7 @@ public:
    // in the SM is shifted from 4.07 MeV 3.98 MeV
    // none of this makes any sense
    // this happend with clang 10.0.0, but not with gcc 9.3.1
-   void set_decay(double width, std::initializer_list<int> products, std::string);
+   void set_decay(double width, std::initializer_list<int> products, const std::string&);
    int get_particle_id() const { return initial_pdg; }
    const Decay& get_decay(std::initializer_list<int> products) const;
    double get_total_width() const { return total_width; }
~~~
gives identical output on my machine. It could also be an issue with the stack size. At least I've seen the strangest behavior if the stack space is exceeded.